crates/bitwarden-ssh/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use rand::SeedableRng;
+use rand_chacha::ChaCha8Rng;
+use ssh_key::{Algorithm, HashAlg, LineEnding};
+
+pub mod error;
+pub mod models;
+
+pub enum KeyAlgorithm {
+    Ed25519,
+    Rsa3072,
+    Rsa4096,
+}
+
+pub fn generate_keypair(
+    key_algorithm: KeyAlgorithm,
+) -> Result<models::SshKey, error::KeyGenerationError> {
+    // sourced from cryptographically secure entropy source, with sources for all targets: https://docs.rs/getrandom
+    // if it cannot be securely sourced, this will panic instead of leading to a weak key
+    let mut rng: ChaCha8Rng = ChaCha8Rng::from_entropy();
+
+    let key = match key_algorithm {
+        KeyAlgorithm::Ed25519 => ssh_key::PrivateKey::random(&mut rng, Algorithm::Ed25519),
+        KeyAlgorithm::Rsa3072 | KeyAlgorithm::Rsa4096 => {
+            let bits = match key_algorithm {
+                KeyAlgorithm::Rsa3072 => 3072,
+                KeyAlgorithm::Rsa4096 => 4096,
+                _ => unreachable!(),
+            };
+
+            let rsa_keypair = ssh_key::private::RsaKeypair::random(&mut rng, bits)
+                .map_err(|e| error::KeyGenerationError::KeyGenerationError(e.to_string()))?;
```

Comment:
**suggestion:** Please import the used error rather than always using the full path.
```suggestion
                .map_err(|e| KeyGenerationError::KeyGenerationError(e.to_string()))?;
```